### PR TITLE
Feature 53639 controller tests

### PR DESCRIPTION
**Pull request by lucashabersaat**

### CODE REVIEW

File: spec/controllers/accounts_controller_spec.rb [hunk 1]
```diff
@@ -32,6 +32,28 @@
       expect(response).to redirect_to teams_path
     end
 
+    it 'shows an error if you look into a team if youre admin and not member of' do
```
Comment (lucashabersaat) [inline spec/controllers/accounts_controller_spec.rb:35]: Admin can look into a team, even if they're not member of it, can't they?
https://github.com/puzzle/cryptopus/wiki/Permissions
Team -> private -> Show Account

**Comment by lucashabersaat** [comment]
The Permission page in the wiki need some additions.
Account action are completely lacking, for instance.

File: spec/controllers/api/accounts_controller_spec.rb [hunk 2]
```diff
@@ -7,8 +7,17 @@
```
Comment (lucashabersaat) [inline spec/controllers/api/accounts_controller_spec.rb:7]: Tests are present multiple times. Caused when I merged this morning?

File: spec/controllers/admin/users_controller_spec.rb [hunk 3]
```diff
@@ -215,6 +248,40 @@
         expect(flash[:error]).to match(/Access denied/)
       end
     end
+
+    describe 'user' do
+
+      it 'cannot update user attributes as users' do
```
Comment (psunix) [inline spec/controllers/admin/users_controller_spec.rb:254]: as users ? user ?

File: spec/controllers/admin/users_controller_spec.rb [hunk 4]
```diff
@@ -215,6 +248,40 @@
         expect(flash[:error]).to match(/Access denied/)
       end
     end
+
+    describe 'user' do
```
Comment (psunix) [inline spec/controllers/admin/users_controller_spec.rb:252]: I think it's a pretty good idea to use this describe blocks to structure the tests by HTTP action.
e.g. describe 'POST update' ....

File: spec/controllers/api/accounts_controller_spec.rb [hunk 5]
```diff
@@ -30,6 +39,32 @@
       expect(account1_json['group']).to eq group.id
     end
 
+    it 'returns account with matching name as admin' do
```
Comment (psunix) [inline spec/controllers/api/accounts_controller_spec.rb:42]: same here for those tests .... there's only member, non-member role for accounts !

File: spec/controllers/api/accounts_controller_spec.rb [hunk 6]
```diff
@@ -100,6 +188,14 @@
 
       expect(result_json['group']).to eq group.id
     end
+
+    it ' doesnot return account for matching tag without cleartext username / password as admin' do
```
Comment (psunix) [inline spec/controllers/api/accounts_controller_spec.rb:192]: typo ...

File: spec/controllers/api/admin/users/role_controller_spec.rb [hunk 7]
```diff
@@ -105,10 +146,8 @@
       end
     end
 
-    context 'as conf_admin' do
```
Comment (psunix) [inline spec/controllers/api/admin/users/role_controller_spec.rb:108]: why not "as conf_admin" ?

as conf_admin do ... sounds pretty natural, doesn't it ?

File: spec/controllers/api/admin/users_controller_spec.rb [hunk 8]
```diff
@@ -32,6 +32,30 @@
       expect(response).to have_http_status(403)
```
Comment (psunix) [inline spec/controllers/api/admin/users_controller_spec.rb:32]: yay, http status check :man_dancing:

File: spec/controllers/api/api_users_controller_spec.rb [hunk 9]
```diff
@@ -5,12 +5,14 @@
 describe Api::ApiUsersController do
   include ControllerHelpers
 
-  before(:each) do
-    login_as(:bob)
+  before(:each) do |test|
+    login_as(:bob) unless test.metadata[:admin_test]
```
Comment (psunix) [inline spec/controllers/api/api_users_controller_spec.rb:9]: didn't know you could do this with rspec! nicely researched guys !
but ... I don't like it ... sry :joy: 
it makes the code not really readable ...
there's also only two roles around api_users .... 1. my api users, 2. not mine
so no need for testing this against the other roles user, admin, conf_admin

File: spec/controllers/api/groups_controller_spec.rb [hunk 10]
```diff
@@ -21,6 +21,21 @@
 
     end
 
+    it 'returns matching groups as admin' do
```
Comment (psunix) [inline spec/controllers/api/groups_controller_spec.rb:24]: this is also just team membership related !

**Pull request closed without merging**